tests/unit/serializer-test.js

```diff
@@ -197,79 +234,100 @@ test('normalizing with synchronous relationships', function(assert) {
+        'replies': {
+          'data': [
+            { 'type': 'reply', 'id': '7' },
+            { 'type': 'reply', 'id': '8' }
+          ]
+        }
+      }
+    }, {
+      'type': 'reply',
+      'id': '7',
+      'attributes': {
+        'body': 'The first reply to the first comment'
+      },
+      'relationships': {}
+    }, {
+      'type': 'reply',
+      'id': '8',
+      'attributes': {
+        'body': 'The second reply to the first comment'
+      },
       'relationships': {}
     }, {
       'type': 'comment',
       'id': '4',
       'attributes': {
         'body': 'The second comment body'
       },
+      'relationships': {
+        'replies': {
+          'data': [
+            { 'type': 'reply', 'id': '9' },
+            { 'type': 'reply', 'id': '10' }
+          ]
+        }
+      }
+    }, {
+      'type': 'reply',
+      'id': '9',
+      'attributes': {
+        'body': 'The first reply to the second comment'
+      },
+      'relationships': {}
+    }, {
+      'type': 'reply',
+      'id': '10',
+      'attributes': {
+        'body': 'The second reply to the second comment'
+      },
       'relationships': {}
     }]
   };
 
   assert.deepEqual(serializer.normalizeResponse(store, postModel, payload, '1', 'findRecord'), expectedNormalization);
 });
-
-test('serializes json api style data to a query usable as an ArgumentSet', function(assert) {
-  let serializer = new Serializer();
-  serializer.container = new ContainerDouble({
-    'transform:string': { serialize: function(v) { return v; } }
-  });
-
-  let expected = {
-    'title': 'The title',
-    'body': 'The body',
-    'authorId': '1',
-    'postIds': ['1']
-  };
-
-  let projectAttrs = {
-    title: 'The title',
-    body: 'The body'
-  };
-
-  let projectRels = {
-    author: { kind: 'belongsTo', key: 'author', data: { id: '1', modelName: 'user' } },
-    posts: { kind: 'hasMany', key: 'posts', data: [{ id: '1', modelName: 'post' }] },
-  };
-
-  assert.deepEqual(serializer.serialize(new SnapshotDouble('project', projectAttrs, projectRels)), expected);
-});
-
-test('serializes json api style data using custom normalizeCase function', function(assert) {
-  let serializer = new Serializer({
-    normalizeCase: Ember.String.underscore
-  });
-  serializer.container = new ContainerDouble({
-    'transform:string': { serialize: function(v) { return v; } }
-  });
-
-  let expected = {
-    'title': 'The title',
-    'body': 'The body',
-    'long_body': 'The long body',
-    'author_id': '1',
-    'post_ids': ['1']
-  };
-
-  let projectAttrs = {
-    title: 'The title',
-    body: 'The body',
-    longBody: 'The long body'
-  };
-
-  let projectRels = {
-    author: { kind: 'belongsTo', key: 'author', data: { id: '1', modelName: 'user' } },
-    posts: { kind: 'hasMany', key: 'posts', data: [{ id: '1', modelName: 'post' }] },
-  };
-
-  assert.deepEqual(serializer.serialize(new SnapshotDouble('project', projectAttrs, projectRels)), expected);
-});
```

Comment:
I don't see that much value in these unit tests now that we have proper integration tests. What do you think about removing them?